File: test_pkg/src/test_node1.cpp
#include "ros/ros.h"

#include "std_msgs/String.h"



void callback_string(const std_msgs::String::ConstPtr& msg){
	
	// ROS_INFO("I heard: [%s]", msg->data.c_str());
	ROS_INFO_STREAM("Receiving... " << msg->data.c_str());

}




int main(int argc, char **argv){

	ros::init(argc, argv, "test_node_1");

	ros::NodeHandle n;

	ros::Subscriber sub = n.subscribe("/topic1", 1000, callback_string);

	ros::Publisher pub = n.advertise<std_msgs::String>("/topic1", 1000);

	ros::Rate r(1); 


    while(ros::ok()){

        std_msgs::String msg;
        msg.data = "hello";
        pub.publish(msg);
		ROS_INFO("Sending...");
		ros::spinOnce();
        r.sleep();
		
    }


}

File: test_pkg/script/test_node1.py
#!/usr/bin/env python3

import rospy

from std_msgs.msg import String

def callback_string(msg):
    rospy.logwarn("I heard %s", msg.data)


if __name__ == '__main__':
    rospy.init_node("test_node_1")
    sub = rospy.Subscriber("/topic1", String, callback_string, queue_size=1000)
    pub = rospy.Publisher("/topic1",String, queue_size=1000)

    r = rospy.Rate(1)

    while not rospy.is_shutdown():
        msg = String()
        msg.data = "hello"
        pub.publish(msg)
        rospy.loginfo("Sending..")

        r.sleep()